### PR TITLE
ENH: added 'site' tag to the GNSS TEC Instrument

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Added 'site' tag to the GNSS TEC Instrument.

# Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality or documentation)

# How Has This Been Tested?

```
import datetime as dt
import pysat
import pysatMadrigal as pymad

stime = dt.datetime(2001, 1, 1)
site = pysat.Instrument(inst_module=pymad.instruments.gnss_tec, tag='site', user='Name', password='email')
site.load(date=stime)
```

## Test Configuration
* Operating system: OSX Big Sur
* Version number: Python 3.8
* Any details about your local setup that are relevant: pysat develop branch

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysatMadrigal/instruments/gnss_tec.py [hunk 1]
```diff
@@ -49,46 +49,46 @@
...
-dname = '{{year:02d}}{{month:02d}}{{day:02d}}'
+dname = '{{month:02d}}{{day:02d}}'
 vname = '.{{version:03d}}'
-supported_tags = {ss: {'vtec': ''.join(['gps', dname, 'g', vname,
+supported_tags = {ss: {'vtec': ''.join(['gps{{year:02d}}', dname, 'g', vname,
+                                        ".{file_type}"]),
+                       'site': ''.join(['site_{{year:04d}}', dname, vname,
                                         ".{file_type}"])}
                   for ss in inst_ids.keys()}
 remote_tags = {ss: {kk: supported_tags[ss][kk].format(file_type='hdf5')
```
Comment (rstoneback) [inline pysatMadrigal/instruments/gnss_tec.py:65]: Tests have an error that I think is related to the difference in file extension, '.h5' vs '.hdf5'
```
files = ['site_20010101.001.h5']
format_str = 'site_{year:04d}{month:02d}{day:02d}.{version:03d}.hdf5'
```

File: pysatMadrigal/instruments/gnss_tec.py [hunk 2]
```diff
@@ -49,46 +49,46 @@
...
 # Instrument methods
 
 
 def init(self):
-    """Initializes the Instrument object with values specific to GNSS TEC
-
-    Runs once upon instantiation.
-
-    """
+    """Initializes the Instrument object with values specific to GNSS TEC."""
```
Comment (rstoneback) [inline pysatMadrigal/instruments/gnss_tec.py:84]:
```suggestion
    """Initialize the Instrument object with values specific to GNSS TEC."""
```

File: pysatMadrigal/instruments/gnss_tec.py [hunk 3]
```diff
@@ -102,10 +102,12 @@ def clean(self):
...
-    if self.tag == "vtec":
-        logger.info("".join(["Data provided at a clean level, further ",
-                             "cleaning may be performed using the ",
-                             "measurement error 'dtec'"]))
+    if self.tag in ["vtec", "site"]:
+        msg = "Data provided at a clean level"
+        if self.tag == "vtec":
+            msg = "".join([msg, ", further cleaning may be performed using ",
+                           "the measurement error 'dtec'"])
+        logger.info(msg)
```
Comment (rstoneback) [inline pysatMadrigal/instruments/gnss_tec.py:110]: brainstorming idea to keep repeated log entries to a minimum. What about setting self.clean_level = None, with a note, in this function so the message is only logged once?

File: pysatMadrigal/instruments/gnss_tec.py (hunk 1 above)
Comment (aburrell) [reply pysatMadrigal/instruments/gnss_tec.py:65]: This was also an issue in the general instrument pull.  Added a fix in the general 'list_remote_files' function.

File: pysatMadrigal/instruments/gnss_tec.py (hunk 3 above)
Comment (aburrell) [reply pysatMadrigal/instruments/gnss_tec.py:110]: That goes against what we've been doing as our standard, so should be brought up at the development meeting for discussion and implementation across all ecosystem packages.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**